File: PHP Projects/CollatzConjecture.php
<?php
declare(strict_types=1);
function steps(int $number, int $steps = 0): int
{
    if ($number > 1) {
        if ($number % 2 == 0) {
            $number = $number/2;
            $steps += 1;
            return steps($number, $steps);
        } else {
            $number = ($number * 3) + 1;
            $steps += 1;
            return steps($number, $steps);
        }
    } elseif ($number == 1) {
        return $steps;
    } else {
        throw new InvalidArgumentException("Only positive numbers are allowed");
    }
}    

File: Python Projects/Python Problems/collatz_conjecture.py
def steps(number, count = 0):
    if number > 1:
        if number % 2 == 0:
            number = number/2
            count += 1
            return steps(number, count)
        else:
            number = (number * 3) + 1
            count += 1
            return steps(number, count)
    elif number == 1:
        return count
    else:
        raise ValueError("Only positive integers are allowed")